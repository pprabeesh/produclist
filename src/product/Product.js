import React from "react";
import { StyledContainer, StyledImage, StyledTags } from "./Product.styles";

const Product = product => {
  return (
    <StyledContainer>
      <StyledImage
        bground={
          process.env.PUBLIC_URL + "/assets/images/" + product.productImage
        }
      />
      <StyledTags />
      <StyledTags />
    </StyledContainer>
  );
};

export default Product;
