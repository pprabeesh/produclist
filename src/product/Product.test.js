import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Product from "./Product";

test("Product renders correctly", () => {
  const product = {
    productImage: "test.jpg",
    isExclusive: true,
    isSale: true,
    productName: "test",
    price: "0.00"
  };
  let container = document.createElement("div");

  renderer.act(() => {
    ReactDOM.render(<Product product={product} />, container);
  });

  expect(container).toMatchSnapshot("Not expanded");
});
