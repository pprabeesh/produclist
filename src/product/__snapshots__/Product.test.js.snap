// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly: Not expanded 1`] = `
<div>
  <div
    class="sc-bwzfXH sc-htpNat eDtUTK sc-bdVaJa iHZvIS"
  >
    <div
      bground="/assets/images/undefined"
      class="sc-bdVaJa sc-bxivhb dxynJq"
    />
    <div
      class="sc-bwzfXH sc-ifAKCX cojtVS sc-bdVaJa iHZvIS"
    />
    <div
      class="sc-bwzfXH sc-ifAKCX cojtVS sc-bdVaJa iHZvIS"
    >
      <div
        class="sc-bdVaJa sc-bZQynM eXVguj"
      />
      <div
        class="sc-bdVaJa sc-gzVnrw dCrIVZ"
      />
    </div>
  </div>
</div>
`;
